pkg/controllers/management/auth/globalroles/enqueue_test.go

```diff
@@ -911,3 +914,173 @@ func Test_namespaceEnqueueGR(t *testing.T) {
 		})
 	}
 }
+func TestClusterRoleEnqueueGRB(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	tests := map[string]struct {
+		obj      runtime.Object
+		grCache  func() mgmtv3.GlobalRoleCache
+		wantKeys []relatedresource.Key
+	}{
+		"enqueue grb if cr contains the label authz.management.cattle.io/grb-owner": {
+			obj: &v1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						grOwnerLabel: "gr",
+					},
+				},
+			},
+			grCache: func() mgmtv3.GlobalRoleCache {
+				mock := fake.NewMockNonNamespacedCacheInterface[*v3.GlobalRole](ctrl)
+				mock.EXPECT().GetByIndex(grSafeConcatIndex, "gr").Return([]*v3.GlobalRole{{
+					ObjectMeta: metav1.ObjectMeta{Name: "gr"},
+				}}, nil)
+				return mock
+			},
+			wantKeys: []relatedresource.Key{
+				{
+					Name: "gr",
+				},
+			},
+		},
+		"don't enqueue grb if cr doesn't contain the label authz.management.cattle.io/grb-owner": {
+			obj: &v1.ClusterRole{},
+			grCache: func() mgmtv3.GlobalRoleCache {
+				return fake.NewMockNonNamespacedCacheInterface[*v3.GlobalRole](ctrl)
+			},
+			wantKeys: nil,
+		},
+	}
+
+	for name, test := range tests {
+		test := test
+		t.Run(name, func(t *testing.T) {
+			g := globalRBACEnqueuer{
+				grCache: test.grCache(),
+			}
+			keys, err := g.clusterRoleEnqueueGR("", "", test.obj)
+			assert.NoError(t, err)
+			assert.Equal(t, test.wantKeys, keys)
+		})
+	}
+}
+
+func TestClusterRoleBindingEnqueueGRB(t *testing.T) {
```

Comment:
There's a couple of test cases that would be good to add:
- Nil obj
- Input object not a clusterRoleBinding
- Error when retrieveing from caches

The test cases also seem to use the old label.

Reply:
added in the latest commit https://github.com/rancher/rancher/pull/44901/commits/723d79e964a44843ce33dce0e3264ece4ac7a5c8